torch/_dynamo/symbolic_convert.py

```diff
@@ -1666,8 +1666,20 @@ def __init__(
 
         vars = list(code_options["co_varnames"])
         vars.extend(x for x in self.cell_and_freevars() if x not in vars)
+
         self.symbolic_locals = collections.OrderedDict(
-            (k, VariableBuilder(self, LocalSource(k))(f_locals[k]))
+            (
+                k,
+                VariableBuilder(
+                    self,
+                    LocalInputSource(
+                        k,
+                        code_options["co_varnames"].index(k)
+                        if k in code_options["co_varnames"]
+                        else None,
+                    ),
+                )(f_locals[k]),
```

Comment:
this mostly looks right, but if you pass a global to a function does it show up in f_locals here? or does that need to also be handled?

Reply:
Let me add a test, good call